Have lexbox API container check fw-headless health

rmunn opened the pull request:
Fix #1185.

Health checks for fw-headless container now go through lexbox-api's health check endpoint.

rmunn commented:
Commit 67897602 implements the HealthCheckConfig idea I mentioned in the PR description.

rmunn commented:
@hahn-kev - I might have done something slightly wrong with registering the health checks in ASP.NET — because when I just tested this in local-dev, the health check returned Unhealthy when I intended for it to return Degraded. I had everything up and running and the health checks were returning Healthy. Then I ran `kubectl delete pod fw-headless-6bd44b6d66-tngql` and immediately re-ran the health checks on http://localhost:5158/api/healthz.

It should have returned "Degraded" according to how we designed this (and I thought I'd implemented it correctly), but instead it returned "Unhealthy". However, I can't find my error; from what I can see, it should have been returning "Degraded" since it couldn't connect to the fw-headless health endpoint. So I need a second pair of eyes on the code to tell me what mistake I'm making.

hahn-kev commented on backend/FwHeadless/Program.cs:
```diff
@@ -35,7 +37,16 @@
+    ResponseWriter = async (context, healthReport) =>
+    {
+        var version = typeof(Program).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "dev";
+        context.Response.Headers["lexbox-version"] = version;
```
I'd rather just add the header to all requests like we do in lexbox api

rmunn replied:
Done in commit f1034055.

hahn-kev reviewed (changes requested):
one minor change, otherwise it looks good

rmunn commented:
@hahn-kev - Unfortunately, you'll need to re-approve. I had to rebase on top of develop to fix a merge conflict in skaffold.yaml. And since my post-rebase was not the same set of changes that you approved (I removed the skaffold.yaml change since we're going to be deleting it soon anyway), that means GitHub doesn't auto-reapply your approval. (For all it knows, the two lines I removed from skaffold.yaml could have broken the whole application!)
